library/psa_crypto.c

```diff
@@ -3467,7 +3467,8 @@ psa_status_t psa_cipher_encrypt( mbedtls_svc_key_id_t key,
     status = psa_driver_wrapper_cipher_encrypt(
         &attributes, slot->key.data, slot->key.bytes,
         alg, local_iv, default_iv_length, input, input_length,
-        output + default_iv_length, output_size - default_iv_length,
+        ( output == NULL ? NULL : output + default_iv_length ),
```

Comment:
Just as a possibility, so we can have something concrete

```suggestion
        NULL_SAFE_PTR_ADD( output, default_iv_length ),
```

which has the property: the words `NULL` and `SAFE` give an indication of WHY this is done; `PTR_ADD` should be sufficiently explanatory and easy to learn; doesn't require repeating `output` or `NULL` and is actually shorter.

The only disadvantage (depending how understandable you find `PTR_ADD`) is that the `+` sign disappears

Reply:
This part is from https://github.com/Mbed-TLS/mbedtls/pull/6609 and I won't change it again here. I'm going to rebase those commits out, in fact, since they're causing confusion. (I hadn't rebased them because there are no merge conflicts and I thought keeping them would help incremental review.)

Reply:
It doesn't look like #6609 touched `psa_crypto.c` - what am I missing?

Reply:
Ah, yes, my bad, only the bignum part was in the earlier PR, not the PSA crypto part. So this is still up for review.

Reply:
okay, why not separate that bit out into an independent PR? Then this one really can be "get the CI running a later/better sanitiser"

Reply:
I originally wanted to have non-regression testing for the issues I'm fixing. But given how hard getting even one of Travis or Jenkins to run on Ubuntu 20.04 is turning out, I'll do that.

Reply:
Tentative name [`mbedtls_buffer_offset`](https://github.com/Mbed-TLS/mbedtls/pull/6648/files#diff-d204d3b703c823d0b6bf79cec59e96cea18691a1cf1149ba32d09ec5c72529b9R72). I made it a function rather than a macro because it's safer, but a downside is that it's typed so there needs to be a separate name for the `const` version.